storage/libradb/src/lib.rs

```diff
@@ -768,6 +768,43 @@ impl LibraDB {
     }
 
     // ================================== Backup APIs ===================================
+
+    /// Gets an iterator which can yield transactions in the range of `[start, end)`.
+    pub fn get_transaction_iter(
+        &self,
+        start_version: Version,
+        end_version: Version,
+    ) -> Box<dyn Iterator<Item = Result<Transaction>>> {
+        struct TxnIter {
+            store: Arc<TransactionStore>,
+            next_version: Version,
+            end_version: Version,
+        }
+
+        impl Iterator for TxnIter {
+            type Item = Result<Transaction>;
+
+            fn next(&mut self) -> Option<Self::Item> {
+                if self.next_version >= self.end_version {
+                    return None;
+                }
+
+                let ret = self
+                    .store
+                    .get_transaction_opt(self.next_version)
```

Comment:
it'll be more efficient to use a DB iter (than point get)?

Reply:
I think that's a good point. I tried a bit and the only issue I'm seeing is to do that the code will roughly become the following and it seems nontrivial to make the lifetime work. Any ideas?

```
struct TxnIter<'a> {
    store: Arc<TransactionStore>,
    inner: SchemaIterator<'a, TransactionSchema>,
}
```

Reply:
Yeah seems no way without unsafe code (one way of using unsafe code is through `owning_ref::OwningHandle`) at this point. 

How about we change the interface? like:

```
let backup_handler = libra_db.get_backup_handler()  // wraps around `Arc<DB>`
let iter = backup_handler.get_txn_iter()  // good until the backup_handler goes out of scope
// use the iter
```